Source/Core/VideoBackends/Metal/MTLUtil.mm

```diff
@@ -0,0 +1,470 @@
+  auto q = MRCTransfer([dev newCommandQueue]);
+  u32 px = 0x11223344;
+  memcpy([buf contents], &px, 4);
+  id<MTLCommandBuffer> cmdbuf = [q commandBuffer];
+  id<MTLBlitCommandEncoder> upload_encoder = [cmdbuf blitCommandEncoder];
+  [upload_encoder copyFromBuffer:buf
+                    sourceOffset:0
+               sourceBytesPerRow:4
+             sourceBytesPerImage:4
+                      sourceSize:MTLSizeMake(1, 1, 1)
+                       toTexture:tex
+                destinationSlice:0
+                destinationLevel:0
+               destinationOrigin:MTLOriginMake(0, 0, 0)];
+  [upload_encoder endEncoding];
+  auto rpdesc = MRCTransfer([MTLRenderPassDescriptor new]);
+  [[rpdesc colorAttachments][0] setTexture:tex];
+  [[rpdesc colorAttachments][0] setLoadAction:MTLLoadActionLoad];
+  [[rpdesc colorAttachments][0] setStoreAction:MTLStoreActionStore];
+  id<MTLRenderCommandEncoder> renc = [cmdbuf renderCommandEncoderWithDescriptor:rpdesc];
+  [renc setRenderPipelineState:pipe];
+  [renc drawPrimitives:MTLPrimitiveTypeTriangle vertexStart:0 vertexCount:3];
+  [renc endEncoding];
+  id<MTLBlitCommandEncoder> download_encoder = [cmdbuf blitCommandEncoder];
+  [download_encoder copyFromTexture:tex
+                        sourceSlice:0
+                        sourceLevel:0
+                       sourceOrigin:MTLOriginMake(0, 0, 0)
+                         sourceSize:MTLSizeMake(1, 1, 1)
+                           toBuffer:buf
+                  destinationOffset:0
+             destinationBytesPerRow:4
+           destinationBytesPerImage:4];
+  [download_encoder endEncoding];
+  [cmdbuf commit];
+  [cmdbuf waitUntilCompleted];
+  u32 outpx;
+  memcpy(&outpx, [buf contents], 4);
+  // Proper fbfetch will double contents, Haswell will return black, and Broadwell will do nothing
+  if (outpx == 0x22446688)
+    return true;  // Skylake+
+  else if (outpx == 0x11223344)
+    return false;  // Broadwell
+  else
+    return false;  // Haswell
+}
+
+void Metal::Util::PopulateBackendInfoFeatures(VideoConfig* config, id<MTLDevice> device)
+{
+  // Initialize DriverDetails first so we can use it later
+  DriverDetails::Vendor vendor = DriverDetails::VENDOR_UNKNOWN;
+  if ([[device name] containsString:@"NVIDIA"])
+    vendor = DriverDetails::VENDOR_NVIDIA;
+  else if ([[device name] containsString:@"AMD"])
+    vendor = DriverDetails::VENDOR_ATI;
+  else if ([[device name] containsString:@"Intel"])
+    vendor = DriverDetails::VENDOR_INTEL;
+  else if ([[device name] containsString:@"Apple"])
+    vendor = DriverDetails::VENDOR_APPLE;
+  NSOperatingSystemVersion cocoa_ver = [[NSProcessInfo processInfo] operatingSystemVersion];
+  double version = cocoa_ver.majorVersion * 100 + cocoa_ver.minorVersion;
+  DriverDetails::Init(DriverDetails::API_METAL, vendor, DriverDetails::DRIVER_APPLE, version,
+                      DriverDetails::Family::UNKNOWN);
+
+#ifdef TARGET_OS_MAC
```

Comment:
`TARGET_OS_MAC` is always defined and is always `1` on all macOS-derived platforms (including iOS). You probably want `#if TARGET_OS_OSX` - see `TargetConditionals.h` from the SDK.